gcloud-java-core/src/main/java/com/google/cloud/BaseServiceException.java

```diff
@@ -187,7 +188,10 @@ protected boolean isRetryable(boolean idempotent, Error error) {
   }
 
   protected boolean isRetryable(boolean idempotent, IOException exception) {
-    return idempotent && exception instanceof SocketTimeoutException;
+    boolean exceptionIsRetryable = exception instanceof SocketTimeoutException
+        || exception instanceof SocketException
+        || "insufficient data written".equals(exception.getMessage());
```

Comment:
Lets hope that a retry indeed fix the "insufficient data written" case (which sounds like a bug in the google-api-java-client, however I am not sure what actually triggers it).